File: engine/engine-model/src/main/scala/tools/packet/ui/ShowOXQuiz.scala
package tools.packet.ui

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowOXQuiz(private var _questionSet: Int, private var _questionId: Int, private var _askQuestion: Boolean) extends PacketInput {
  def questionSet: Int = _questionSet

  def questionId: Int = _questionId

  def askQuestion: Boolean = _askQuestion

  override def opcode(): SendOpcode = SendOpcode.OX_QUIZ
}

File: engine/engine-model/src/main/java/tools/packet/ui/ShowOXQuiz.java
package tools.packet.ui;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowOXQuiz(Integer questionSet, Integer questionId, Boolean askQuestion) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.OX_QUIZ;
   }
}